### PR TITLE
[SYCL][Doc] Add release notes for May release

### PR DESCRIPTION
**Pull request by romanovvlad**
Coauthored by @mibintc, @kbobrovs, @DenisBakhvalov, @s-kanaev, @AlexeySachkov, @mdtoguchi

### CODE REVIEW

File: sycl/ReleaseNotes.md [hunk 1]
```diff
@@ -1,3 +1,218 @@
+# May'21 release notes
+
+## New features
```
Comment (DenisBakhvalov) [inline sycl/ReleaseNotes.md:3]: I suggest adding a note:
- [ESIMD] Allowed ESIMD and regular SYCL kernels to coexist in the same translation unit and in the same program. The `-fsycl-explicit-simd` option is no longer required for compiling ESIMD code and was deprecated. DPCPP RT implicitly appends `-vc-codegen` compile option for ESIMD images.

There were ~25 commits, so it's not possible to list them all. :)

File: sycl/ReleaseNotes.md [hunk 2]
```diff
@@ -1,3 +1,218 @@
...
+   [12e8549] [c2f211a]
+ - Added support for
+   [SYCL_INTEL_local_memory extension](doc/extensions/LocalMemory/SYCL_INTEL_local_memory.asciidoc)
+   [5a66fcb] [9a734f6]
+ - Documented [Level Zero backend](doc/extensions/LevelZeroBackend/LevelZeroBackend.md)
+   [8994e6d]
+
+## Improvements
+### SYCL Compiler
+ - Added support for math builtins: `fmax`, `fmin`, `isinf`, `isfinite`,
```
[the remark this answers is not in the document]
Comment (mibintc) [reply sycl/ReleaseNotes.md:35]: It's ok with me if you want to hyphenate, but in the clang User Manual that's mostly not hyphenated. eg "standard builtin functions".  Do we have a style manual for guidance?

[the remark this answers is not in the document]
Comment (sergey-semenov) [reply sycl/ReleaseNotes.md:35]: I'm not aware of any such style manual, mostly I just wanted to keep the wording consistent across this doc (L187 uses the hyphenated version). FWIW, SYCL spec also uses "built-in", I think it's a good idea to have our release notes follow suit.

**Review by sergey-semenov — APPROVED** [review]

**Review by DenisBakhvalov — DISMISSED** [review]

File: sycl/ReleaseNotes.md [hunk 3]
```diff
@@ -1,3 +1,219 @@
...
+ - Improved range rounding heuristics [7efb692]
+ - Added `get_backend` methods to SYCL classes [ee7e99f]
+ - Added `sycl::sub_group::load` and `sycl::sub_group::store` versions that
+   take raw pointers [248f550]
+ - Enabled caching of devices in `sycl::device` interoperability constructors
+   [d3aeb4a]
+ - Added a warning on using SYCL 1.2.1 OpenCL interoperability API when
+   compiling in SYCL 2020 mode. It can be suppressed by defining
+   `SYCL2020_DISABLE_DEPRECATION_WARNINGS` macro or by setting `-sycl-std=2017`
+   or `-sycl-std=1.2.1` [a249316]
```
Comment (pvchupin) [inline sycl/ReleaseNotes.md:89]:
```suggestion
   `SYCL2020_DISABLE_DEPRECATION_WARNINGS` macro` [a249316]
```
Please stop recommending -sycl-std as workarounds!

File: sycl/ReleaseNotes.md [hunk 4]
```diff
@@ -1,3 +1,219 @@
...
+ - Added `__LIBSYCL_[MAJOR|MINOR|PATCH]_VERSION` macros, see
+   [PreprocessorMacros](doc/PreprocessorMacros.md) for more information
+   [9f3a74c]
+ - Added support for SYCL 2020 reductions with `read_write` access mode to
+   reduction variables [733d5e3]
+ - Added support for SYCL 2020 reductions with
+   `sycl::property::reduction::initialize_to_identity` property [3473c1a]
+ - Implemented zero argument version of `sycl::buffer::reinterpret()` for
+   SYCL 2020 [c0c3c80]
+ - Implemented [Matrix Programming Extension for DPC++](https://github.com/intel/llvm/blob/49b6749ea9175ae250b718c04d71af4ccfecc06c/sycl/doc/extensions/Matrix/dpcpp-joint-matrix.asciidoc) [35db973]
```
Comment (pvchupin) [inline sycl/ReleaseNotes.md:28]: Is it full support or partial? @dkhaldi 
Written like a full support, but I'm not sure it is.

File: sycl/ReleaseNotes.md [hunk 5]
```diff
@@ -1,3 +1,219 @@
...
+ - Added a diagnostic on attempt to compile a SYCL application without dynamic
+   C++ RT on Windows [d4180f4]
+ - Added support for `Queue Order Properties` extension for Level Zero [50005c7]
+ - Improved plugin discovery mechanism - if a plugin fails to initialize others
+   will be discovered anyway [d513074]
+ - Added support for `sycl::info::partition_affinity_domain::numa` in Level
+   Zero plugin [2ba8e05]
+### Documentation
+ - Added notes on SYCL 2020 status and runtime headers C++14 standard
+   requirement [a05f5de] [fed2d47]
```
Comment (pvchupin) [inline sycl/ReleaseNotes.md:103]:
```suggestion
 - Added notes on SYCL 2020 status and runtime headers C++17 standard
   requirement [a05f5de] [fed2d47]
```

File: sycl/ReleaseNotes.md [hunk 6]
```diff
@@ -1,3 +1,219 @@
...
+   open up the device and lock it to that process since the runtime needs
+   to query the actual device to obtain that information.
+ - On Windows, DPC++ compiler enforces using dynamic C++ runtime for
+   application linked with SYCL library by:
+    - linking with msvcrt[d].dll when `-fsycl` switch is used;
+    - emitting an error on attempts to compile a program with static C++ RT
+      using `-fsycl` and `/MT` or `/MTd`.
+   That protects you from complicated runtime errors caused by C++ objects
+   crossing sycl[d].dll boundary and not always handled properly by different
+   versions of C++ RT used on app and sycl[d].dll sides.
```
Comment (pvchupin) [inline sycl/ReleaseNotes.md:203]: Should we remove it and translate that to documented -fsycl behavior on Windows?
It's not known issue it seems, it's rather how -fsycl switch is expected to work.
@v-klochkov

File: sycl/ReleaseNotes.md [hunk 7]
```diff
@@ -1,3 +1,219 @@
...
+ - The format of the object files produced by the compiler can change between
+   versions. The workaround is to rebuild the application.
+ - Using `sycl::program`/`sycl::kernel_bundle` API to refer to a kernel defined
+   in another translation unit leads to undefined behavior
+ - Linkage errors with the following message:
+   `error LNK2005: "bool const std::_Is_integral<bool>" (??$_Is_integral@_N@std@@3_NB) already defined`
+   can happen when a SYCL application is built using MS Visual Studio 2019
+   version below 16.3.0 and user specifies `-std=c++14` or `/std:c++14`.
```
Comment (pvchupin) [inline sycl/ReleaseNotes.md:211]: We don't support C++14 switches, so can be removed I think.

File: sycl/ReleaseNotes.md [hunk 8]
```diff
@@ -1,3 +1,219 @@
...
+ - Printing internal defines isn't supported on Windows [50628db]
+ - Group algorithms for MUL/AND/OR/XOR cannot be enabled for group scope due to
+   SPIR-V limitations, and are not enabled for sub-group scope yet as the
+   SPIR-V version isn't automatically raised from 1.1 to 1.3 [96da39e]
```
Comment (pvchupin) [inline sycl/ReleaseNotes.md:215]: Haven't we raised spir-v at #3267

**Review by pvchupin — CHANGES_REQUESTED** [review]

File: sycl/ReleaseNotes.md (hunk 4 above)
Comment (dkhaldi) [reply sycl/ReleaseNotes.md:28]: @pvchupin The document does not say full support. Only base features are supported. Mainly the query interface is not there yet. Also, this is only AOT on CPU AMX.

I would suggest reword the release note to:
- Added an initial AOT implementation of the experimental matrix extension on the CPU device to target AMX hardware. Base features are supported.

Comment (pvchupin) [reply sycl/ReleaseNotes.md:28]: Sounds good to me. Thanks @dkhaldi for suggestion.

File: sycl/ReleaseNotes.md (hunk 5 above)
Comment (romanovvlad) [reply sycl/ReleaseNotes.md:103]: The commit which does this is not included in the release.

File: sycl/ReleaseNotes.md (hunk 8 above)
Comment (romanovvlad) [reply sycl/ReleaseNotes.md:215]: @v-klochkov Could you please confirm that this "known issue" can be removed?

File: sycl/ReleaseNotes.md (hunk 7 above)
Comment (romanovvlad) [reply sycl/ReleaseNotes.md:211]: See https://github.com/intel/llvm/pull/3590#discussion_r620187774

File: sycl/ReleaseNotes.md (hunk 8 above)
Comment (v-klochkov) [reply sycl/ReleaseNotes.md:217]: Yes, it can be removed. Thank you.

File: sycl/ReleaseNotes.md (hunk 6 above)
Comment (v-klochkov) [reply sycl/ReleaseNotes.md:205]: I think this statement may be removed from 'known issues' list.

File: sycl/ReleaseNotes.md (hunk 5 above)
Comment (pvchupin) [reply sycl/ReleaseNotes.md:105]: I would just remove this item from the list to avoid confusion.

File: sycl/ReleaseNotes.md (hunk 7 above)
Comment (pvchupin) [reply sycl/ReleaseNotes.md:213]: Ok. Can cleanup in the next one.

**Review by pvchupin — APPROVED** [review]
Pls. report testing issue.

**Comment by romanovvlad** [comment]
> 
> 
> Pls. report testing issue.

Done.

**Pull request merged**